src/mscorlib/src/System/String.cs

```diff
@@ -51,13 +45,23 @@ public sealed class String : IComparable, ICloneable, IConvertible, IEnumerable
         //NOTE NOTE NOTE NOTE
         //These fields map directly onto the fields in an EE StringObject.  See object.h for the layout.
         //
-        [NonSerialized]private int  m_stringLength;
+        [NonSerialized] private int m_stringLength;
 
-        [NonSerialized]private char m_firstChar;
+        // For empty strings, this will be '\0' since
+        // strings are both null-terminated and length prefixed
+        [NonSerialized] private char m_firstChar;
+        
+        // For empty strings, this will be null due to padding.
+        // The start of the string (not including sync block pointer)
+        // is the method table pointer + string length, which takes up
+        // 8 bytes on 32-bit, 12 on x64. For empty strings the null
+        // terminator immediately follows, leaving us with an object
+        // 10/14 bytes in size. Since everything needs to be a multiple
+        // of 4/8, this will get padded and zeroed out.
+        
+        // For one-char strings this will be the null terminator.
+        [NonSerialized] private char m_secondChar;
```

Comment:
@jkotas Gotten rid of the field and moved the check after `fixed`. The results appear to have slowed down slightly for a difference in the second char (0.020... to 0.023/5...) but it's still faster than the previous version.